socorro/signature/rules.py

```diff
@@ -260,8 +263,13 @@ def _do_generate(self, source_list, hang_type, crashed_thread, delimiter=' | '):
         # Get all the relevant frame signatures.
         new_signature_list = []
         for a_signature in source_list:
-            # If the signature matches the irrelevant signatures regex, skip to the next frame.
-            if self.irrelevant_signature_re.match(a_signature):
+            # We want to match against the function signature or any of the parts of the
+            # signature in the case where there are specifiers and return types.
+            a_signature_parts = a_signature.split() + [a_signature]
```

Comment:
So this is the meat of the fix. We used to get frames like this:

```
mozilla::net::nsHttpConnectionMgr::Shutdown
```

and now we're getting frames like this:

```
static void mozilla::net::nsHttpConnectionMgr::Shutdown
```

The "static" part is a specifier. The "void" part is the return type. I think we could build a list of all the specifiers we're seeing and that list would be stable over time and we wouldn't have to futz with it much. The return types are a different story. There are some common ones like "int" and "bool" and "void", but then there are a lot of other ones that are defined in the Firefox code like "JSObject*".

So instead of stripping specifiers and return types, we split the signature into parts and make sure to add the whole thing so then the irrelevant and prefix things try to match any of the strings in this list:

```
['static', 'void', 'mozilla::net::nsHttpConnectionMgr::Shutdown', 'static void mozilla::net::nsHttpConnectionMgr::Shutdown']
```

Adding the whole signature at the end allows for the `<name omitted>` case in the prefix list which splits poorly.